apps/concurrent/modules/user_input_parsing.pl

```diff
@@ -85,11 +86,13 @@
     ;   put_assoc(Action, Assoc, Value, Domain).
 
 parse_parts([], Domain, Domain).
-parse_parts([H|T], Assoc, Domain) :-
+parse_parts([H | T], Assoc, Domain) :-
     normalize_space(atom(Line), H),
-    parse_sentence(Line, Action, Value),
-    add_to_domain(Action, Value, Assoc, Domain_Tmp),
-    parse_parts(T, Domain_Tmp, Domain).
+    (Line = ''
```

Comment:
This code makes the parser resistant to `\n` characters at the beginning, end and in the middle of the domain text.